### ISSUE TITLE
Consider deployment story for static web apps/spa

### ISSUE THREAD

**Issue by timheuer** [dotnet/aspire#2711]
As some front-end apps get added for AppHost orchestration and thus also deployment, currently Dockerfile is the recommended (only?) approach. This will translate into a container resource for deployment manifest.

We should consider how to model specifically SPA-type front-ends in the AppHost that also would translate to new deployment endpoints more tailored to static web apps like storage, Azure SWA, etc.

**Comment by kjbetz** [comment]
I agree. I'm working on a current project that I would like to deploy an Angular SPA to an Azure Static Web App. The API deploying as a Container App and the DB provisioned as Postgres Flexible Server should be good. Would love to see this picked up.

Or/And also another scenario maybe... being able to build frontend and put into public folder of web app before it gets built and deployed as Azure Container App.

**Comment by onionhammer** [comment]
This would be super useful. SWA supports API backends, so it could map '/api/' to a container app, or potentially multiple container apps using ACA's 'routes' feature.

I could imagine Aspire could tackle deployments with SWA cli being installed with npm

**Comment by KennethHoff** [comment]
Recently started researching SWAs for a potential customer and nowadays the first thing I always do is research how to do that thing in Aspire. Given the work recently around deployment and static assets I was surprised SWAs weren't implemented. I realize the functions part is quite complicated, but the provisioning bit as well as the static file hosting feels like it wouldn't be too complicated to get working.
